### DISCUSSION TITLE
Submit button with value

### DISCUSSION THREAD

**Discussion by mysuf** [react-hook-form/react-hook-form#6996] [Q&A] [answered]
Hi, I just noticed that `handleSubmit` is ignoring submit value, f.e. `<button name="submit" value="1">Submit</button>`. I would expect property `submit` present in data object passed to submit callback. Can anybody provide example if registering is needed? Thanks

**Comment by bluebill1049** [comment]
- register is needed
- or use a hidden input 
- or use a custom register at useEffect

**Comment by ritikbanger** [comment, marked as answer]
Hello @mysuf,

`handleSubmit` is just a handler, it will only pass the registered inputs to a particular function you give to the `handleSubmit` handler.

Like `handleSubmit={onSubmit}` will call the function onSubmit on submit button.

The best way is to use a register in your fields either as `ref={register({ required: true, maxLength: 50 })}` or `{...register("id",{ required: true, maxLength: 50 })}` as per the version you are using.

Register do register inputs of your fields and boom you can get the data.

For reference: 
`  const onSubmit = async (data: any) => {
    console.log('Data: ', data)`
